tests/Tokenizer/Analyzer/AttributeAnalyzerTest.php

```diff
@@ -129,4 +130,286 @@ public static function provideIsAttributeCases(): iterable
 
         yield [true, '<?php #[Foo("(")] class Bar {}'];
     }
+
+    /**
+     * @requires PHP 8.0
+     *
+     * @dataProvider provideGetAttributeDeclarationsCases
+     *
+     * @param array{0: string, 1: list<AttributeAnalysis>} $expected
+     */
+    public function testGetAttributeDeclarations(string $code, int $startIndex, array $expected): void
+    {
+        $tokens = Tokens::fromCode($code);
+        $attributeDeclarationAnalyses = AttributeAnalyzer::collect($tokens, $startIndex);
+
+        self::assertSame(
+            serialize($expected),
+            serialize($attributeDeclarationAnalyses),
+        );
+    }
```

Comment:
```suggestion
     * @param array{0: string, 1: list<AttributeAnalysis>} $expectedAnalyses
     */
    public function testGetAttributeDeclarations(string $code, int $startIndex, array $expectedAnalyses): void
    {
        $tokens = Tokens::fromCode($code);
        $actualAnalyses = AttributeAnalyzer::collect($tokens, $startIndex);

        foreach ($expectedAnalyses as $expectedAnalysis) {
            self::assertSame(T_ATTRIBUTE, $expectedAnalysis->getStartIndex());
            self::assertSame(CT::T_ATTRIBUTE_CLOSE, $expectedAnalysis->getEndIndex());
        }

        self::assertSame(
            serialize($expectedAnalyses),
            serialize($actualAnalyses ),
        );
    }
```

Reply:
Similarly in `testGetAttributeDeclarations81()`.

Reply:
Done